hermes/bin/src/runtime_extensions/hermes/cardano/host.rs

```diff
@@ -393,8 +399,8 @@ impl HostTransaction for HermesRuntimeContext {
         &mut self,
         self_: wasmtime::component::Resource<Transaction>,
     ) -> wasmtime::Result<Option<TxnHash>> {
-        let mut app_state = STATE.transaction.get_app_state(self.app_name())?;
-        let object = app_state.get_object(&self_)?;
+        let app_state = STATE.transaction.get_app_state_readonly(self.app_name())?;
```

Comment:
Could you consider a minor rename?
`get_app_state()` -> `get_app_state_mut()`
and then
`get_app_state_readonly()` -> `get_app_state()`